### PR TITLE
Add hal.h from the prima driver

### PR DESCRIPTION
**Pull request by kvalo**
This pull requests adds hal.h from the prima driver. The benefit from this is that we use the same terms as the prima driver.

I did a lot of cleanup but there's still few pendings pending, for example endian support, few typedefs and structure field names. I'll clean those later.

### CODE REVIEW

**Comment by KrasnikovEugene** [comment]
enums like eSIR_WEP_STATIC  must be renamed

**Comment by KrasnikovEugene** [comment]
enums like eSTA_11ac must be removed since HW is not supporting 11ac

**Comment by KrasnikovEugene** [comment]
suggest also to change variable name to something more linux friendli e.g.  ucMaxBssids change to max_bssids

**Comment by KrasnikovEugene** [comment]
Also suggest to change PE_SUMMARY_STATS_INFO to something else because PE is a similar to SIR(one of the layers in prima driver) and not valid for wcn36xx any more

**Comment by kvalo** [comment]
In most cases we should use WCN36XX_HAL where it’s not excessive

KrasnikovEugene notifications@github.com writes:

> Also suggest to change PE_SUMMARY_STATS_INFO to something else because PE is a
> similar to SIR(one of the layers in prima driver) and not valid for wcn36xx any
> more

Maybe I should use WCN36XX_HAL_ and wcn36xx_hal_ prefixes everywhere?

## 

Kalle Valo

File: hw.h [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+ * AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+ * DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
+ * PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+ * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ * PERFORMANCE OF THIS SOFTWARE.
+ */
+/*
+ * Copyright (c) 2012, The Linux Foundation. All rights reserved.
+ *
+ * Previously licensed under the ISC license by Qualcomm Atheros, Inc.
```
Comment (KrasnikovEugene) [inline hw.h]: copyrights twice. probably to excessive;)

Comment (kvalo) [inline hw.h]: KrasnikovEugene notifications@github.com writes:

> In hw.h:
> 
> > - \* above copyright notice and this permission notice appear in all
> > - \* copies.
> > - *
> > - \* THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
> > - \* WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
> > - \* WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
> > - \* AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
> > - \* DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
> > - \* PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
> > - \* TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
> > - \* PERFORMANCE OF THIS SOFTWARE.
> > - _/
> >   +/_
> > - \* Copyright (c) 2012, The Linux Foundation. All rights reserved.
> > - *
> > - \* Previously licensed under the ISC license by Qualcomm Atheros, Inc.
> 
> copyrights twice. probably to excessive;)

I have already removed hw.h from my branch already, we just needed one
define from hw.h.

## 

Kalle Valo

**Comment by KrasnikovEugene** [comment]
I suggest to remove this.

#define IS_MCC_SUPPORTED_BY_HOST (!!(halMsg_GetHostWlanFeatCaps(MCC)))
#define IS_SLM_SESSIONIZATION_SUPPORTED_BY_HOST (!!(halMsg_GetHostWlanFeatCaps(SLM_SESSIONIZATION)))
#define IS_FEATURE_SUPPORTED_BY_HOST(featEnumValue) (!!halMsg_GetHostWlanFeatCaps(featEnumValue))
#define IS_WLANACTIVE_OFFLOAD_SUPPORTED_BY_HOST (!!(halMsg_GetHostWlanFeatCaps(WLANACTIVE_OFFLOAD)))

**Comment by KrasnikovEugene** [comment]
Ok, there is so many changes already so we have to merge this and start to use it. Let's just file issues when we found something.

**Pull request merged**